File: api/schemas/session.py
from datetime import datetime
from typing import Literal
from pydantic import Field
from .common import StrictModel
from .risk import RiskLevel, SegmentResult

SessionStatus = Literal["created", "active", "stopping", "completed", "failed"]

class SessionContext(StrictModel):
    call_origin: str | None = Field(default=None, max_length=100)
    contact_match: bool | None = None
    transaction_amount: float | None = Field(default=None, ge=0)
    transaction_type: str | None = Field(default=None, max_length=100)
    privileged_action: bool | None = None
    historical_fraud_indicator: bool | None = None
    new_device: bool | None = None
    unusual_time: bool | None = None
    location_anomaly: bool | None = None

class CreateSessionRequest(StrictModel):
    speaker_id: str | None = Field(default=None, min_length=1, max_length=128, pattern=r"^[A-Za-z0-9_.-]+$")
    context: SessionContext = Field(default_factory=SessionContext)

class CreateSessionResponse(StrictModel):
    session_id: str
    status: SessionStatus
    created_at: datetime

class SessionResponse(StrictModel):
    session_id: str
    status: SessionStatus
    speaker_id: str | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None
    segments_processed: int
    current_risk_score: float
    current_risk_level: RiskLevel

class StopSessionResponse(StrictModel):
    session_id: str
    status: SessionStatus
    segments_processed: int
    highest_risk_score: float
    final_risk_level: RiskLevel

class SessionResultsResponse(StrictModel):
    session_id: str
    results: list[SegmentResult]
    limit: int
    offset: int
    total: int

File: frontend/src/types/session.ts
import type { RiskLevel, SegmentResult } from './risk'
export type SessionStatus = 'created'|'active'|'stopping'|'completed'|'failed'
export const TERMINAL_SESSION_STATUSES: readonly SessionStatus[] = ['completed','failed']
export const isTerminalSession = (status:SessionStatus) => TERMINAL_SESSION_STATUSES.includes(status)
export interface SessionContext { call_origin?:string; contact_match?:boolean; transaction_amount?:number; transaction_type?:string; privileged_action?:boolean }
export interface CreateSessionInput { speaker_id?:string; context:SessionContext }
export interface CreatedSession { session_id:string; status:SessionStatus; created_at:string }
export interface Session { session_id:string; status:SessionStatus; speaker_id:string|null; created_at:string; started_at:string|null; completed_at:string|null; segments_processed:number; current_risk_score:number; current_risk_level:RiskLevel }
export interface SessionResults { session_id:string; results:SegmentResult[]; limit:number; offset:number; total:number }
export interface StopResult { session_id:string; status:SessionStatus; segments_processed:number; highest_risk_score:number; final_risk_level:RiskLevel }
